src/main/java/github/scarsz/discordsrv/listeners/PlayerAdvancementDoneListener.java

```diff
@@ -59,21 +64,46 @@ public void onPlayerAdvancementDone(PlayerAdvancementDoneEvent event) {
             return;
         }
 
+        TextChannel channel = DiscordSRV.getPlugin().getMainTextChannel();
+        String channelName = channel.getName();
+        String message = LangUtil.Message.PLAYER_ACHIEVEMENT.toString();
+        Player player = event.getPlayer();
+        Advancement advancement = event.getAdvancement();
+
         // turn "story/advancement_name" into "Advancement Name"
-        String rawAdvancementName = event.getAdvancement().getKey().getKey();
+        String rawAdvancementName = advancement.getKey().getKey();
         String advancementName = Arrays.stream(rawAdvancementName.substring(rawAdvancementName.lastIndexOf("/") + 1).toLowerCase().split("_"))
                 .map(s -> s.substring(0, 1).toUpperCase() + s.substring(1))
                 .collect(Collectors.joining(" "));
 
-        String discordMessage = LangUtil.Message.PLAYER_ACHIEVEMENT.toString()
+        AchievementMessagePreProcessEvent preEvent = DiscordSRV.api.callEvent(new AchievementMessagePreProcessEvent(channelName, message, player, advancementName));
```

Comment:
Cancellation should be checked

Reply:
Changed this to be consistent with your other uses (consistent API). So the results of it being cancelled are passed into the post-processing event, then the checking of the message being cancelled is done after the post-processing event.